packages/vm/lib/evm/precompiles/index.ts

```diff
@@ -29,24 +49,101 @@ const precompiles: Precompiles = {
 }
 
 const precompileAvailability: PrecompileAvailability = {
-  '0000000000000000000000000000000000000001': 'chainstart',
-  '0000000000000000000000000000000000000002': 'chainstart',
-  [ripemdPrecompileAddress]: 'chainstart',
-  '0000000000000000000000000000000000000004': 'chainstart',
-  '0000000000000000000000000000000000000005': 'byzantium',
-  '0000000000000000000000000000000000000006': 'byzantium',
-  '0000000000000000000000000000000000000007': 'byzantium',
-  '0000000000000000000000000000000000000008': 'byzantium',
-  '0000000000000000000000000000000000000009': 'istanbul',
+  '0000000000000000000000000000000000000001': {
+    type: PrecompileAvailabilityCheck.Hardfork,
+    param: 'chainstart',
+  },
+  '0000000000000000000000000000000000000002': {
+    type: PrecompileAvailabilityCheck.Hardfork,
+    param: 'chainstart',
+  },
+  [ripemdPrecompileAddress]: { type: PrecompileAvailabilityCheck.Hardfork, param: 'chainstart' },
+  '0000000000000000000000000000000000000004': {
+    type: PrecompileAvailabilityCheck.Hardfork,
+    param: 'chainstart',
+  },
+  '0000000000000000000000000000000000000005': {
+    type: PrecompileAvailabilityCheck.Hardfork,
+    param: 'byzantium',
+  },
+  '0000000000000000000000000000000000000006': {
+    type: PrecompileAvailabilityCheck.Hardfork,
+    param: 'byzantium',
+  },
+  '0000000000000000000000000000000000000007': {
+    type: PrecompileAvailabilityCheck.Hardfork,
+    param: 'byzantium',
+  },
+  '0000000000000000000000000000000000000008': {
+    type: PrecompileAvailabilityCheck.Hardfork,
+    param: 'byzantium',
+  },
+  '0000000000000000000000000000000000000009': {
+    type: PrecompileAvailabilityCheck.Hardfork,
+    param: 'istanbul',
+  },
+  '000000000000000000000000000000000000000a': {
+    type: PrecompileAvailabilityCheck.EIP,
+    param: 'EIP2537',
+  },
+  '000000000000000000000000000000000000000b': {
+    type: PrecompileAvailabilityCheck.EIP,
+    param: 'EIP2537',
+  },
+  '000000000000000000000000000000000000000c': {
+    type: PrecompileAvailabilityCheck.EIP,
+    param: 'EIP2537',
+  },
+  '000000000000000000000000000000000000000d': {
+    type: PrecompileAvailabilityCheck.EIP,
+    param: 'EIP2537',
+  },
+  '000000000000000000000000000000000000000f': {
+    type: PrecompileAvailabilityCheck.EIP,
+    param: 'EIP2537',
+  },
+  '000000000000000000000000000000000000000e': {
+    type: PrecompileAvailabilityCheck.EIP,
+    param: 'EIP2537',
+  },
+  '0000000000000000000000000000000000000010': {
+    type: PrecompileAvailabilityCheck.EIP,
+    param: 'EIP2537',
+  },
+  '0000000000000000000000000000000000000011': {
+    type: PrecompileAvailabilityCheck.EIP,
+    param: 'EIP2537',
+  },
+  '0000000000000000000000000000000000000012': {
+    type: PrecompileAvailabilityCheck.EIP,
+    param: 'EIP2537',
+  },
 }
 
-function getPrecompile(address: string, common: Common): PrecompileFunc {
+function getPrecompile(address: string, common: Common, vm: VM): PrecompileFunc {
```

Comment:
Note: this new `vm` parameter can be removed again once we have `EIP` support moved to `Common`.

Reply:
Yes, should fix once that is merged in